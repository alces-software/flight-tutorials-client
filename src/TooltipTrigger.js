// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
 *
 * This file is part of Flight Common.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';
import { OverlayTrigger, Tooltip } from 'react-bootstrap';

type PropTypes = {
  children : React$Element<*>,
  tooltip? : string | React$Element<*>,
  className? : string,
};

const TooltipTrigger = ({ children, tooltip, className }: PropTypes) => {
  if (tooltip == null || tooltip === '') {
    return <div style={{ display: 'inline-block' }}>{children}</div>;
  }
  return (
    <OverlayTrigger
      overlay={<Tooltip
        id="tooltip-trigger"
        className={className}
      >
        {tooltip}
      </Tooltip>}
      placement="top"
    >
      {children}
    </OverlayTrigger>
  );
};

export default TooltipTrigger;
