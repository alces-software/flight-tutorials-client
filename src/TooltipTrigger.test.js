// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';

import TooltipTrigger from './TooltipTrigger';

it('renders without crashing', () => {
  const node = document.createElement('div');
  render(<TooltipTrigger>My children</TooltipTrigger>, node);
});

const variants = {
  'with tooltip': {
    children: <div>My children</div>,
    tooltip: 'My tooltip',
  },
  'with empty tooltip': {
    children: <div>My children</div>,
    tooltip: '',
  },
  'without tooltip': {
    children: <div>My children</div>,
  },
  'with className': {
    children: <div>My children</div>,
    className: 'myClass',
  },
  'without className': {
    children: <div>My children</div>,
  },
};

Object.keys(variants).forEach((name) => {
  it(`renders correctly ${name}`, () => {
    const props = variants[name];
    const tree = renderer.create(
      <TooltipTrigger {...props} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
