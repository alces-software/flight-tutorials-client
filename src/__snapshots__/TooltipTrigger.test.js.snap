// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with className 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div>
    My children
  </div>
</div>
`;

exports[`renders correctly with empty tooltip 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div>
    My children
  </div>
</div>
`;

exports[`renders correctly with tooltip 1`] = `
<div
  onBlur={[Function]}
  onClick={null}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  My children
</div>
`;

exports[`renders correctly without className 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div>
    My children
  </div>
</div>
`;

exports[`renders correctly without tooltip 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div>
    My children
  </div>
</div>
`;
